### PR TITLE
Developer docs site redesign

### LINKED ISSUE #389: Update style, layout of our doc
Update the template and CSS as needed to use Concordium colors for notes, warnings, etc.

Do this after release of the new web site. Get brand book.

Redo layout of landing page and documentation pages, particularly long and messy side nav.

### PR DESCRIPTION
**Pull request by dg-concordium**
## Purpose

Modernize and clean up our documentation site for developers and users.

## Changes

Lots. :-)

## Checklist

- [X] My code follows the style of this project.
- [X] The code compiles without warnings.
- [X] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [X] I accept the above linked CLA.

Closes #389

### CODE REVIEW

**Review by jens-concordium — COMMENTED** [review]
Some general remarks / remarks I didn't have a specific place to note:

1. Should clicked links maybe not change color? It looks a bit weird in some places, e.g. the landing page:

image

2. Some links and clickable text pieces can be quite difficult to see. I've noted it some places in the review, but I just wanted to make a general remark about it as well.
image

3. If the landing page is scaled just right, the text in the boxes can exceed the box area:
image

4. In the "mobile size" version of the landing page, the boxes don't have any white space around them. Not super important, but I think it would look nice with just a bit on the right and left.

5. Is it intentional that people have to go through the sys reqs, before they can find a guide on how to run a node in the different platforms? It takes quite a few clicks to get to the right guide. It’s also hard to find the guides via the menu on the left in itself. It helps with the links from the landing page and the top menu though. I've also noted at the specific page, that it might be nice with a bit more explanation than just sys reqs, if that page will be the "root" of the run-a-node-guides.

File: source/mainnet/net/nodes/run-node-ubuntu.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
+.. _run-node-ubuntu-testnet:
+
+=============================================
+Run a node on a server with Ubuntu on Testnet
```
[the remark this answers is not in the document]
Comment (dg-concordium) [reply source/mainnet/net/nodes/run-node-ubuntu.rst:4]: It has to do with all of the sub-headings. The tabs cannot handle the heading levels in Sphinx.

File: source/mainnet/_templates/index.html [hunk 2]
```diff
@@ -4,10 +4,11 @@
 <head>
     <meta charset="utf-8" />
     <title>Concordium Documentation</title>
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css" integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous">
```
[the remark this answers is not in the document]
Comment (dg-concordium) [reply source/mainnet/_templates/index.html:7]: I think we are covered under the CC-BY license.

File: source/shared/net/nodes/node-requirements.rst [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
 
-=================================
+========================
+Run a node on Concordium
+========================
+
+When you run a node on the Concordium blockchain, you are participating in the Concordium blockchain by validating transactions.
+
+Nodes are required for most things, from using a Concordium wallet to baking. You can choose to run a node yourself or you can have a third-party provider run a node for you. If you choose to run the node yourself, you can
+choose from four different platforms to run your node: Ubuntu, Windows, MacOS, or Docker/Linux.
```
Comment (jens-concordium) [inline source/shared/net/nodes/node-requirements.rst:10]: Would it make sense to have these link to the different guides?

**Review by jens-concordium — APPROVED** [review]
There was just the one comment about maybe linking to the node guides

File: source/shared/net/nodes/node-requirements.rst (hunk 3 above)
Comment (dg-concordium) [reply source/shared/net/nodes/node-requirements.rst:10]: They are linked further down in the same topic.

**Pull request merged**